distribution/lib/Standard/Base/0.0.0-dev/src/Data/Index_Sub_Range.enso

```diff
@@ -1,11 +1,21 @@
-from Standard.Base import all
+import project.Data.Any.Any
+import project.Data.Numbers.Integer
 
-import Standard.Base.Random
-import Standard.Base.Runtime.Ref
+## Imported this way as then imports the extension methods.
+from project.Data.Range import Range
 
-# TODO Dubious constructor export
-from project.Data.Index_Sub_Range.Index_Sub_Range import all
-from project.Data.Index_Sub_Range.Index_Sub_Range export all
+import project.Data.Vector.Vector
+
+import project.Random
+import project.Runtime.Ref
+
+import project.Math
+
+from project.Data.Boolean import Boolean, True, False
+from project.Error.Common import Error, Panic, Index_Out_Of_Bounds_Error_Data, Illegal_Argument_Error_Data
+
+from project.Data.Index_Sub_Range.Index_Sub_Range import First, Last
+from project.Data.Index_Sub_Range.Index_Sub_Range export First, Last
```

Comment:
Why do we keep re-exporting First and Last here?

Reply:
Kept for simplicity - we use these a lot and the shorthand seem worthwhile.

Reply:
Should we not re-export them in `Standard.Base.Main` then? I don't think the re-export from the more 'internal-facing' `Index_Sub_Range` module is worth it. We should be importing the type not the module anyway - so having First/Last still on the module may tempt someone to wrongly import the module instead of the type.

I'd remove it from here. But I'm more than fine from keeping a re-export at the toplevel of the library.

Reply:
Don't believe it works unless exported from the module itself.

Reply:
Have amended to this (didn't think it works but apparently does).